### DISCUSSION TITLE
Steps component - change step programatically

### DISCUSSION THREAD

**Discussion by DSzczygiel** [chakra-ui/chakra-ui#9330] [Q&A] [answered]
I'm trying to figure out how to change step programatically in Chakra UI v3 `Steps` component.  In Chakra v2 there was `useSteps()` hook which had `nextStep`, `prevStep` functions. In my use case, I have multi-step form and when user clicks _Next_ button, I trigger the validation and only if form is valid, I call the `nextStep()` function which loads another step of that form. I don't know how to achieve same functionality using a `StepsNextTrigger` component.

Do you have any ideas? Thanks!

**Comment by segunadebayo** [comment, marked as answer]
Control it with `step` and `onStepChange` on `Steps.Root`. https://chakra-ui.com/docs/components/steps
